Hybrid support, fixes and updates

captain5050 opened the pull request:
Add hybrid support to the download_and_gen.sh script.
Fix some python3 issues
Update to the latest 01.org events.

captain5050 committed:
Fix python3 issue

captain5050 committed:
Turn json-to-perf-json into a module

This allows hybrid-json-to-perf-json to use the module rather than to
invoke the command. This simplifies some build rules in environments
like bazel.

captain5050 committed:
Fix BDW column header

BDX is renamed to BDX/BDW-DE and BDW/BDW-DE is renamed to BDW by
download_and_gen.sh. Fix the column header name for ADL.

captain5050 committed:
Hybrid support

Also update to latest 01.org files.

ZhengjunXing0 commented on download_and_gen.sh:
```diff
@@ -192,6 +219,38 @@ do
+  echo -e "\\n\\n\\nProcessing hybird core TMA metrics for $core\\n\\n\\n"
+  python3 extract-tma-metrics.py \
+    --memory \
+    --cstate \
```
Only one of "cpu_core" and "cpu_atom" can enable "--cstate", if both are enabled, there should be two same "cstate" metrics. I suggest only enabling the "--cstate" for the later PMU "cpu_atom".

captain5050 replied:
Thanks, done.

ZhengjunXing committed:
Fix pmu name for hybrid system

When generating metrics for "cpu_core" in the  hybrid system, the pmu
name should be "cpu_core".

"cpu_atom" (GRT) no ":c" in the metrics formulas, so it doesn't need
the fix.

Signed-off-by: Zhengjun Xing <zhengjun.xing@linux.intel.com>

ZhengjunXing committed:
Add perf-uncore-events-bdx.csv and perf-uncore-events-hsx.csv

perf-uncore-events-bdx.csv based on BDX uncore events v17
https://download.01.org/perfmon/BDX/

perf-uncore-events-hsx.csv based on HSX uncore events v22
https://download.01.org/perfmon/HSX/

Signed-off-by: Zhengjun Xing <zhengjun.xing@linux.intel.com>

captain5050 committed:
Don't write empty MetricGroups

ZhengjunXing0 commented:
Thanks, applied.
